src/backend/marsha/core/views.py

```diff
@@ -474,6 +478,123 @@ class VideoView(BaseLTIView):
+        )
+
+        if liveregistration.get_generate_salted_hmac() == key:
+
+            session_id = str(uuid.uuid4())
+            cache_key = f"app_data|direct_access|{self.model.__name__}|{video_pk}"
+            app_data = cache.get(cache_key)
+
+            if app_data is None:
+                app_data = self._get_base_app_data()
+                app_data.update(
+                    {
+                        "resource": self.serializer_class(
+                            liveregistration.video,
+                            context={
+                                "is_admin": False,
+                                "user_id": liveregistration.lti_id,
+                                "session_id": session_id,
+                            },
+                        ).data,
+                        "state": "success",
+                        "player": settings.VIDEO_PLAYER,
+                        "uploadPollInterval": settings.FRONT_UPLOAD_POLL_INTERVAL,
+                    }
+                )
+                cache.set(cache_key, app_data, settings.APP_DATA_CACHE_DURATION)
+
+            jwt_token = AccessToken()
+            jwt_token.payload["resource_id"] = str(liveregistration.video.id)
+            jwt_token.payload["user"] = {
+                "email": liveregistration.email,
+            }
+
+            if liveregistration.is_from_lti_connection:
+
+                jwt_token.payload["consumer_site"] = str(
+                    liveregistration.consumer_site.id
+                )
+                jwt_token.payload["context_id"] = str(
+                    liveregistration.video.playlist.lti_id
+                )
+                jwt_token.payload["roles"] = ["student"]
+                jwt_token.payload["user"].update(
+                    {
+                        "id": liveregistration.lti_user_id,
+                        "username": liveregistration.username,
+                    }
+                )
+
+            else:
+
+                jwt_token.payload["user"].update(
+                    {"anonymous_id": str(liveregistration.anonymous_id)}
+                )
+                jwt_token.payload["roles"] = [NONE]
+
+            jwt_token.payload.update(
+                {
+                    "session_id": session_id,
+                    "locale": define_locales(None),
+                    "permissions": {"can_access_dashboard": False, "can_update": False},
+                    "maintenance": settings.MAINTENANCE_MODE,
+                }
+            )
+            app_data.update(
+                {
+                    "jwt": str(jwt_token),
+                }
+            )
+
+            return self.render_to_response(
+                {
+                    "app_data": json.dumps(app_data),
+                    "external_javascript_scripts": settings.EXTERNAL_JAVASCRIPT_SCRIPTS,
+                    "static_base_url": f"{settings.STATIC_URL}js/build/",
+                }
+            )
```

Comment:
This code can be extracted in a private method, accepting `app_data` in parameter and returning the response. Then this new method can also be used by `get_public_data` and `get_context_data`